### PR TITLE
only reverse indices once

### PR DESCRIPTION
**Pull request by calellowitz**
for the tests

### CODE REVIEW

**Comment by calellowitz** [comment]
@proteusvacuum cc: @ctsims 
This should fix the purge algorithm to not take forever by stopping it from constantly recalculating the reversed indices.

**Comment by ctsims** [comment]
Structurally this approach looks correct to me, but am probably not the best person for a deep review.

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 1]
```diff
@@ -750,6 +750,14 @@ class SimplifiedSyncLog(AbstractSyncLog):
     closed_cases = SetProperty(unicode)
     extensions_checked = BooleanProperty(default=False)
 
+    _purged_cases = set()
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: Is this meant to be a class-level mutable set? Would it work to set this to `None` instead so the class attribute is immutable? Seems like bad things would happen if this set instance somehow got an item added to it, and as far as I can tell this instance is never used.

**Review by millerdev — COMMENTED** [review]
Would like to see the class attribute changed to an immutable value, but if this is really urgent it appears that it will work as is, so could be merged.

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 2]
```diff
@@ -750,6 +750,14 @@ class SimplifiedSyncLog(AbstractSyncLog):
     closed_cases = SetProperty(unicode)
     extensions_checked = BooleanProperty(default=False)
 
+    _purged_cases = None
+
+    @property
+    def purged_cases(self):
+        if not self._purged_cases:
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: nitpick: could now change this to `if self._purged_cases is None:` so it does not continually create a new `set` each time `self.purged_cases` is accessed while the set is empty.

**Review by millerdev — APPROVED** [review]

**Pull request merged**